File: backend/src/main/java/com/radar/prospection/repository/MissionRepository.java
package com.radar.prospection.repository;

import com.radar.prospection.domain.Mission;
import com.radar.prospection.domain.MissionStatus;
import com.radar.prospection.domain.Source;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@Repository
public interface MissionRepository extends JpaRepository<Mission, Long> {

    Optional<Mission> findBySourceAndExternalId(Source source, String externalId);

    boolean existsBySourceAndExternalId(Source source, String externalId);

    List<Mission> findByStatusOrderByFitScoreDesc(MissionStatus status);

    List<Mission> findByFitScoreGreaterThanEqualOrderByFitScoreDesc(int minScore);

    @Query("SELECT m FROM Mission m WHERE m.status = 'NEW' ORDER BY m.detectedAt DESC")
    List<Mission> findPendingAnalysis();

    @Query("SELECT m FROM Mission m WHERE m.detectedAt >= :since ORDER BY m.fitScore DESC NULLS LAST")
    List<Mission> findRecentMissions(LocalDateTime since);

    @Query("SELECT COUNT(m) FROM Mission m WHERE m.source = :source AND m.detectedAt >= :since")
    long countBySourceSince(Source source, LocalDateTime since);

    List<Mission> findByFitScoreIsNull();

    List<Mission> findByFavoriteTrueOrderByFitScoreDesc();
}

File: backend/src/main/kotlin/com/radar/prospection/repository/MissionRepository.kt
package com.radar.prospection.repository

import com.radar.prospection.domain.Mission
import com.radar.prospection.domain.MissionStatus
import com.radar.prospection.domain.Source
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.time.LocalDateTime

@Repository
interface MissionRepository : JpaRepository<Mission, Long> {
    fun findBySourceAndExternalId(source: Source, externalId: String): Mission?
    fun existsBySourceAndExternalId(source: Source, externalId: String): Boolean
    fun findByStatusOrderByFitScoreDesc(status: MissionStatus): List<Mission>
    fun findByFitScoreGreaterThanEqualOrderByFitScoreDesc(minScore: Int): List<Mission>
    fun findByFitScoreIsNull(): List<Mission>
    fun findByFavoriteTrueOrderByFitScoreDesc(): List<Mission>

    @Query("SELECT m FROM Mission m WHERE m.status = 'NEW' ORDER BY m.detectedAt DESC")
    fun findPendingAnalysis(): List<Mission>

    @Query("SELECT m FROM Mission m WHERE m.detectedAt >= :since ORDER BY m.fitScore DESC NULLS LAST")
    fun findRecentMissions(since: LocalDateTime): List<Mission>

    @Query("SELECT COUNT(m) FROM Mission m WHERE m.source = :source AND m.detectedAt >= :since")
    fun countBySourceSince(source: Source, since: LocalDateTime): Long
}
